### PR TITLE
Skipped autoupdate for user during login. See #2804.

### PR DESCRIPTION
**Pull request by normanjaeckel**

### CODE REVIEW

**Comment by normanjaeckel** [comment]
@ostcar Please test if this fits your needs.

File: openslides/users/models.py [hunk 1]
```diff
@@ -188,6 +188,16 @@ def __str__(self):
         # Return result
         return name
 
+    def save(self, *args, **kwargs):
+        """
+        Overridden method to skip autoupdate if only last_login field was
+        updated as it is done during login.
+        """
+        update_fields = kwargs.get('update_fields')
+        if update_fields is not None and len(update_fields) == 1 and update_fields[0] == 'last_login':
```
Comment (ostcar) [inline openslides/users/models.py]: Proposal:
```python
if update_fields == ['last_login']:
    ....
```

**Comment by ostcar** [comment]
This fixes only have of my problem. But the second have is maybe another issue (but was fixed with my pull request).

When I try to login 500 users nearly at the same time, it is possible to get this traceback:

```
Internal Server Error: /users/login/
Traceback (most recent call last):
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/channels/handler.py", line 237, in process_exception_by_middleware
    return super(AsgiHandler, self).process_exception_by_middleware(exception, request)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/rest_framework/views.py", line 474, in dispatch
    response = self.handle_exception(exc)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/rest_framework/views.py", line 434, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/rest_framework/views.py", line 471, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/ossi/src/openslides/openslides/users/views.py", line 174, in post
    auth_login(self.request, self.user)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/contrib/auth/__init__.py", line 130, in login
    user_logged_in.send(sender=user.__class__, request=request, user=user)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 191, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/contrib/auth/models.py", line 25, in update_last_login
    user.save(update_fields=['last_login'])
  File "/home/ossi/src/openslides/openslides/users/models.py", line 199, in save
    return super().save(*args, **kwargs)
  File "/home/ossi/src/openslides/openslides/utils/models.py", line 74, in save
    return_value = super().save(*args, **kwargs)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/contrib/auth/base_user.py", line 80, in save
    super(AbstractBaseUser, self).save(*args, **kwargs)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/db/models/base.py", line 796, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/db/models/base.py", line 833, in save_base
    update_fields=update_fields, raw=raw, using=using)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 191, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/home/ossi/src/openslides/openslides/utils/search.py", line 134, in index_add_instance
    with AsyncWriter(index.get_or_create_index()) as writer:
  File "/home/ossi/src/openslides/openslides/utils/search.py", line 75, in get_or_create_index
    if path != 'ram' and exists_in(path):
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/whoosh/index.py", line 136, in exists_in
    ix = open_dir(dirname, indexname=indexname)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/whoosh/index.py", line 123, in open_dir
    return FileIndex(storage, schema=schema, indexname=indexname)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/whoosh/index.py", line 421, in __init__
    TOC.read(self.storage, self.indexname, schema=self._schema)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/whoosh/index.py", line 623, in read
    stream = storage.open_file(tocfilename)
  File "/home/ossi/.virtualenvs/openslides/lib/python3.6/site-packages/whoosh/filedb/filestore.py", line 504, in open_file
    f = StructFile(open(self._fpath(name), "rb"), name=name, **kwargs)
```

It seems, that this commit https://github.com/OpenSlides/OpenSlides/commit/7f1b894db425a3d33692ccae31d352bc81821122 did not fix this PR https://github.com/OpenSlides/OpenSlides/issues/2198

But until this is fixed or the signal is disabled, it is not possible to run the performance script with different users.

File: openslides/users/models.py (hunk 1 above)
Comment (normanjaeckel) [inline openslides/users/models.py]: Done.

**Comment by normanjaeckel** [comment]
The tests fail because `test_admin` tests runs exactly 3 db queries more because of this patch. I do not understand why. @ostcar Any idea?

**Comment by normanjaeckel** [comment]
Test are solved.

**Comment by normanjaeckel** [comment]
The whoosh problem is assigned to another issue (#2865).

**Pull request merged**